lib/faraday/utils.rb

```diff
@@ -95,7 +95,14 @@ def to_hash() ::Hash.new.update(self) end
 
       def parse(header_string)
         return unless header_string && !header_string.empty?
-        header_string.split(/\r\n/).
+
+        headers = header_string.split(/\r\n/)
+
+        # Find the last set of response headers.
+        start_index = headers.rindex { |x| x.match(/^HTTP\//) }
```

Comment:
@bartoszkopinski can you please explain better where you would add the `.to_i`?
But please keep in mind that I'm not willing to add ugly patches to Faraday only to make it compatible with an outdated gem.
The solution in that case would be to either fix the gem (you can always fork it if it's dead), or monkey-patch tin your project